docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/CgovCoreTwigExtensions.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Drupal\cgov_core;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\image\Entity\ImageStyle;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Extend Drupal's Twig_Extension class.
+ */
+class CgovCoreTwigExtensions extends \Twig_Extension {
+
+  /**
+   * Entity Type Manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current Request object.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request
+   */
+  protected $requestStack;
+
+  /**
+   * Constructs a new CgovNavigationManager class.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, RequestStack $requestStack) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->requestStack = $requestStack;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getName() {
+    return 'cgov_core.CgovCoreTwigExtensions';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFunctions() {
+    return [
+      new \Twig_SimpleFunction('get_enclosure', [$this, 'getEnclosure'], ['is_safe' => ['html']]),
+      new \Twig_SimpleFunction('get_list_description', [$this, 'getListDescription'], ['is_safe' => ['html']]),
+    ];
+  }
+
+  /**
+   * Generate <enclosure url='x' length='9' type='mime/type' /> tag from NID.
+   *
+   * Call this function with {{ get_enclosure(nid) }}
+   *
+   * @param int $nid
+   *   Node ID to create enclosure tag from.
+   *
+   * @return string
+   *   generated enclosure tag.
+   */
+  public function getEnclosure($nid) {
+    // Set $debug=TRUE to show displayed image source in comments.
+    $debug = FALSE;
+
+    // Image style to use when rendering images.
+    $thumbnail_image_style = 'cgov_thumbnail';
+
+    // Load the node.
+    $node = $this->entityTypeManager->getStorage('node')->load($nid);
+    if (!$node) {
+      return FALSE;
+    }
+
+    // Check for available images to display.
```

Comment:
Does RSS follow the same rules as promo content where the promotional image is prioritized over the article image? And if so, I'd say update the condition to an if-else so that we don't need to create two images.

Reply:
Yes, it does.  I generally prefer the 'set a default and replace it conditionally' approach is it reduces the chances of developers missing an edge-case with the if-then-else, is a little easier to follow (at least I think it is), and is effectively just as efficient.  But if you want me to change it,  I will.